src/RadixDlt.NetworkGateway.Abstractions/RadixGlobalAddress.cs

```diff
@@ -0,0 +1,87 @@
+ * any distributed ledger nor in respect the functioning or value of any tokens which may
+ * be created stored or transferred using the Work. The Licensor does not warrant that the
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using System.Diagnostics;
+
+namespace RadixDlt.NetworkGateway.Abstractions;
+
+[DebuggerDisplay("{ToString()}")]
+public sealed record RadixGlobalAddress
+{
+    private readonly string _address;
+
+    private RadixGlobalAddress(string address)
+    {
+        _address = address;
+    }
+
+    public static implicit operator string(RadixGlobalAddress ra) => ra._address;
+
+    public static implicit operator RadixGlobalAddress(string text) => new RadixGlobalAddress(text);
```

Comment:
Just thinking out loud: isn't it too easy to mix `EntityGlobalAddress` with other types when we use implicit casting and no guard checks whatsoever? I mean... having method like `MyMethod(EntityId eid, EntityGlobalAddress ga)` we can still call it as `.MyMethod("abc_111111", "abcdef")` or `.MyMethod("abcdef", "abc_111")`.

1. Maybe we should use explicit casting when converting from string to wrapper type?
2. Maybe we could add some super lightweight sanity checks when creating wrapper types? Think `input.All(c => c  >= 'a' and <= 'f' or >= '0' <= '9')` for `EntityId` and `input.Contains('_') && input.All(c => c >= 'a' and <= 'z' or c >= '0' and <= '9')`?
3. Maybe those checks could be only used for implicit casts?

Reply:
**Ad. 1 and 2)** I think, you're right. Even when migrating the code I made a mistake once and simply passed invalid string. I think your idea to create explicit cast and add sanity checks there is a good idea.

Do you think we should create separate sanity check logic or reuse some existing one i.e https://github.com/radixdlt/babylon-gateway/blob/4f34959dfb43a5200847cf7bae8a323bbefdec92/src/RadixDlt.NetworkGateway.Abstractions/Addressing/Bech32Codec.cs#L248 ? 

Logic there is pretty simple so it shouldn't affect performance that much and we'd not duplicate validation logic, what do you think?

**Ad. 3)** to be honest I'd probably leave explicit cast only.

Reply:
OK, so explicit cast from string to wrapper type, implicit cast from wrapper type to string and no extra checks? Sounds good to me.

Reply:
what about reusing that validation method?

Reply:
It is potentially bit heavy, I'd say not to use it when data comes from trusted source (our internal databases etc.), i.e. let's skip that.